src/RA_AchievementSet.cpp

```diff
@@ -420,148 +423,142 @@ BOOL AchievementSet::LoadFromFile(GameID nGameID)
-        g_pCurrentGameData->SetGameID(nGameID);
+    std::ifstream ifile{ sFilename };
 
-        if (m_nSetType == Local)
-        {
-            const char EndLine = '\n';
-            char buffer[4096];
-            unsigned long nCharsRead = 0;
+    // in this case we do have to check
+    if (!ifile.is_open())
+        return FALSE;
 
-            //	Get min ver:
-            _ReadTil(EndLine, buffer, 4096, &nCharsRead, pFile);
-            //	UNUSED at this point? TBD
 
-            //	Get game title:
-            _ReadTil(EndLine, buffer, 4096, &nCharsRead, pFile);
-            if (nCharsRead > 0)
-            {
-                buffer[nCharsRead - 1] = '\0';	//	Turn that endline into an end-string
+    //	Store this: we are now assuming this is the correct checksum if we have a file for it
+    g_pCurrentGameData->SetGameID(nGameID);
 
-                //	Loading Local from file...?
-                g_pCurrentGameData->SetGameTitle(buffer);
-            }
-
-            while (!feof(pFile))
+    if (m_nSetType == Local)
+    {
+        std::string buffer;
+        buffer.reserve(4096); 
+
+        //	Get min ver:
+        ifile.getline(buffer.data(), buffer.capacity());
+        auto nCharsRead{ ifile.gcount() };  //	UNUSED at this point? TBD 
+                  
+        //	Get game title:
+        ifile.getline(buffer.data(), buffer.capacity());
+        nCharsRead = ifile.gcount();
+        //	Loading Local from file...?
+        if (nCharsRead > 0)
+            g_pCurrentGameData->SetGameTitle(buffer.data());
+
+        while (!ifile.eof())
+        {
+            ifile.getline(buffer.data(), buffer.capacity());
+            if (nCharsRead = ifile.gcount(); nCharsRead > 0)
             {
-                ZeroMemory(buffer, 4096);
-                _ReadTil(EndLine, buffer, 4096, &nCharsRead, pFile);
-                if (nCharsRead > 0)
+                buffer = buffer.data(); 
+                if (buffer.front() == 'L')
                 {
-                    if (buffer[0] == 'L')
-                    {
-                        //SKIP
-                        //RA_Leaderboard lb;
-                        //lb.ParseLine(buffer);
+                    //SKIP
+                    //RA_Leaderboard lb;
+                    //lb.ParseLine(buffer);
 
-                        //g_LeaderboardManager.AddLeaderboard(lb);
-                    }
-                    else if (isdigit(buffer[0]))
-                    {
-                        Achievement& newAch = g_pLocalAchievements->AddAchievement();
-                        buffer[nCharsRead - 1] = '\0';	//	Turn that endline into an end-string
-
-                        newAch.ParseLine(buffer);
-                    }
+                    //g_LeaderboardManager.AddLeaderboard(lb);
+                }
+                else if (isdigit(buffer.front()))
+                {
+                    Achievement& newAch = g_pLocalAchievements->AddAchievement();
```

Comment:
If achievements are ID'd copying should be deleted (also meaning they can't be used as pointers but can be cheaply passed by value), we should consider using `unordered_map `if you prefer speed over order or `map `if you prefer order over speed. We can easily use` quick sort` from `std::algorithm` to so sort it if needed. This type of copying technically works even with deleted copying since you are copying a reference (which the object at that address is always const), otherwise you should just using move semantics.